Bug: test_add_rack failure due to unmatched Ethernet136 status

### Is it platform specific

mellanox

### Importance or Severity

High

### Description of the bug

This ticket is used to track fix https://github.com/sonic-net/sonic-mgmt/pull/24658

The configlet.test_add_rack.test_add_rack test has been flaky on Cisco-8102-C64 and Mellanox SN4600C/SN4700 platforms, while consistently passing on Arista. The failure manifests as PORT|Ethernet136 retaining admin_status: up in CONFIG_DB and APP_DB after the T0 removal patch, causing the DB comparison to fail across all 8 retry cycles.

### Steps to Reproduce

Run test_add_rack on specific platforms

### Actual Behavior and Expected Behavior

AssertionError: DB compare failed after adding T0 via generic patch updater

### Relevant log output

```shell
duthost = , skip_load = True, hack_apply = False

    def generic_patch_rm_t0(duthost, skip_load=False, hack_apply=False):
        # Load config with T0
        #
        if not skip_load:
            duthost.copy(src=os.path.join(orig_db_dir, "config_db.json"),
                         dest="/etc/sonic/config_db.json")
            config_reload(duthost, wait=RELOAD_WAIT_TIME, start_bgp=True)
    
        if hack_apply:
            # Hack: TODO: Before removing port, patch updater need to ensure
            # the port is down. Until then bring it down explicitly.
            #
            for link in tor_data["links"]:
                tor_ifname = link["local"]["sonic_name"]
                duthost.shell("config interface shutdown {}".format(tor_ifname))
                do_pause(PAUSE_INTF_DOWN, "pause upon i/f {} shutdown before add patch".format(tor_ifname))
    
        patch_files = _list_patch_files(patch_rm_t0_dir)
    
        CMD = CMD_APPLY_HACK if hack_apply else CMD_APPLY
    
        for fl in patch_files:
            sonic_fl = os.path.join("/etc/sonic", fl)
            duthost.copy(src=os.path.join(patch_rm_t0_dir, fl), dest=sonic_fl)
            duthost.shell(CMD.format(sonic_fl), module_ignore_errors=True)
    
            # HACK: TODO: There are scenarios, where it applies patch and still consider as failed
            # "...Error: After applying patch to config, there are still some parts not updated\n"
            # The above error is possible, if some control plane component is making an update.
            # We can ignore rc, as DB comp is the final check. So skip it.
            # assert res["rc"] == 0, "Failed to apply patch"
    
        # Manual shutdown needed because the removal of admin_status won't operate shutdown. It will
        # by default keep the previous admin_status state. Thus making app-db comparison fail.
        for link in tor_data["links"]:
            tor_ifname = link["local"]["sonic_name"]
            duthost.shell("config interface shutdown {}".format(tor_ifname))
            do_pause(PAUSE_INTF_DOWN, "pause upon i/f {} shutdown before add patch".format(tor_ifname))
    
>       assert wait_until(DB_COMP_WAIT_TIME, 20, 0, db_comp, duthost, patch_rm_t0_dir,
                          no_t0_db_dir, "generic_patch_rm_t0"), \
            "DB compare failed after adding T0 via generic patch updater"
E       AssertionError: DB compare failed after adding T0 via generic patch updater

CMD        = 'config apply-patch -v {}'
duthost    = 
fl         = 'patch_0_all.json'
hack_apply = False
link       = {'local': {'alias': 'Ethernet136', 'sonic_name': 'Ethernet136'}, 'remote': 'Ethernet1'}
patch_files = ['patch_0_all.json']
skip_load  = True
sonic_fl   = '/etc/sonic/patch_0_all.json'
tor_ifname = 'Ethernet136'

configlet/util/generic_patch.py:231: AssertionError
```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/configlet/util/generic_patch.py
+++ b/tests/configlet/util/generic_patch.py
@@ -128,6 +128,26 @@ def _list_patch_files(patch_dir):
     return sorted(fnmatch.filter(os.listdir(patch_dir), "patch_[0-9]_*.json"))
 
 
+def _restore_interface_admin_status(duthost, target_dir):
+    with open(os.path.join(target_dir, "config_db.json"), "r") as stream:
+        target_ports = json.load(stream).get("PORT", {})
+
+    for link in tor_data["links"]:
+        interface = link["local"]["sonic_name"]
+            port_config = target_ports.get(interface)
+            assert isinstance(port_config, dict), \
+                "Missing PORT configuration for interface {}".format(interface)
+
+            admin_status = port_config.get("admin_status", "down")
+            assert admin_status in ("up", "down"), \
+                "Unexpected admin_status '{}' for interface {}".format(admin_status, interface)
+
+        action = "startup" if admin_status == "up" else "shutdown"
+        pause = PAUSE_INTF_UP if admin_status == "up" else PAUSE_INTF_DOWN
+        duthost.shell("config interface {} {}".format(action, interface))
+        do_pause(pause, "pause upon i/f {} {} after remove patch".format(interface, action))
+
+
 def generic_patch_add_t0(duthost, skip_load=False, hack_apply=False):
     # Load config w/o T0
     #
@@ -221,13 +241,10 @@ def generic_patch_rm_t0(duthost, skip_load=False, hack_apply=False):
         # We can ignore rc, as DB comp is the final check. So skip it.
         # assert res["rc"] == 0, "Failed to apply patch"
 
-    # Manual shutdown needed because the removal of admin_status won't operate shutdown. It will
-    # by default keep the previous admin_status state. Thus making app-db comparison fail.
-    for link in tor_data["links"]:
-        tor_ifname = link["local"]["sonic_name"]
-        duthost.shell("config interface shutdown {}".format(tor_ifname))
-        do_pause(PAUSE_INTF_DOWN, "pause upon i/f {} shutdown before add patch".format(tor_ifname))
+    # Applying a remove patch can preserve a transient interface state when admin_status
+    # is removed or reordered. Restore the state captured from the no-T0 configuration.
+    _restore_interface_admin_status(duthost, no_t0_db_dir)
 
     assert wait_until(DB_COMP_WAIT_TIME, 20, 0, db_comp, duthost, patch_rm_t0_dir,
                       no_t0_db_dir, "generic_patch_rm_t0"), \
-        "DB compare failed after adding T0 via generic patch updater"
+        "DB compare failed after removing T0 via generic patch updater"

--- a/tests/configlet/util/generic_patch.py
+++ b/tests/configlet/util/generic_patch.py
@@ -134,13 +134,13 @@ def _restore_interface_admin_status(duthost, target_dir):
 
     for link in tor_data["links"]:
         interface = link["local"]["sonic_name"]
-            port_config = target_ports.get(interface)
-            assert isinstance(port_config, dict), \
-                "Missing PORT configuration for interface {}".format(interface)
+        port_config = target_ports.get(interface)
+        assert isinstance(port_config, dict), \
+            "Missing PORT configuration for interface {}".format(interface)
 
-            admin_status = port_config.get("admin_status", "down")
-            assert admin_status in ("up", "down"), \
-                "Unexpected admin_status '{}' for interface {}".format(admin_status, interface)
+        admin_status = port_config.get("admin_status", "down")
+        assert admin_status in ("up", "down"), \
+            "Unexpected admin_status '{}' for interface {}".format(admin_status, interface)
 
         action = "startup" if admin_status == "up" else "shutdown"
         pause = PAUSE_INTF_UP if admin_status == "up" else PAUSE_INTF_DOWN

